test/extended/apiserver/access.go

```diff
@@ -0,0 +1,79 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/image"
+)
+
+var _ = g.Describe("[Conformance][sig-api-machinery][Feature:APIServer] kube-apiserver should be accessible via", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithPodSecurityLevel("apiserver", admissionapi.LevelPrivileged)
+
+	for description, apiPath := range map[string]string{
+		"service network": "$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT",
+		"api-int":         "api-int",
+		"api-ext":         "api-ext",
+	} {
+		g.It(fmt.Sprintf("%s endpoint", description), func() {
+			// external controlplane topology doesn't have master nodes
+			controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if *controlPlaneTopology == configv1.ExternalTopologyMode {
+				g.Skip("ExternalControlPlaneTopology doesn't have master node kubeconfigs")
+			}
+			// get external/internal URLs
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if apiPath == "api-int" || apiPath == "api-ext" {
+				externalAPIUrl, err := url.Parse(infra.Status.APIServerURL)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if apiPath == "api-ext" {
+					apiPath = externalAPIUrl.Host
+				} else {
+					apiPath = strings.Replace(externalAPIUrl.Host, "api.", "api-int.", 1)
+				}
```

Comment:
Is `api-int.*` always going to resolve to the internal API server endpoint? 

Would pulling the internal API server URL using [`infra.Status.APIServerInternalURL` ](https://github.com/openshift/api/blob/c1fdeb0788c16659238d93ec45ce2e798c14eb88/config/v1/types_infrastructure.go#L89-L93) be more reflective of the flow that a user/client may attempt to take if they were to look at the `Infrastructure` API?

Reply:
Yes, good point. iiuc Hypershift may use a different URL for internal URL

Reply:
Fixed in e1fd7a84533